src/main/java/com/lambdatest/tunnel/Tunnel.java

```diff
@@ -211,81 +215,78 @@ public static void clearTheFile() throws IOException {
 
-        String filePath = ".lambdatest/tunnelprocs/" + t1.port + ".txt";
-        Path pathOfFile = Paths.get(filePath);
-  
-        if (!Files.exists(pathOfFile)) {
-          System.out.println("File does not exist, cannot stop tunnel");
-          return; // Exit the method if the file does not exist
-        }
         try {
-  
-          setTunnelIdFromJson(filePath);
-          boolean stopTunnelAPI = Boolean.parseBoolean(System.getenv("STOP_TUNNEL_API"));
-  
-          String deleteEndpoint = stopTunnelAPI ?
-            "https://ts.lambdatest.com/v1.0/stop/" + tunnelID :
-            "http://127.0.0.1:" + String.valueOf(infoAPIPortValue) + "/api/v1.0/stop";
-  
-          HttpDelete httpDelete = new HttpDelete(deleteEndpoint);
-  
-          if (!stopTunnelAPI) {
-  
-            CloseableHttpClient httpclient = HttpClients.createDefault();
-            HttpResponse response = httpclient.execute(httpDelete);
-            BufferedReader br = new BufferedReader(new InputStreamReader((response.getEntity().getContent())));
-  
-            // Throw runtime exception if status code isn't 200
-            if (response.getStatusLine().getStatusCode() != 200) {
-              throw new RuntimeException("Failed : HTTP error code : " + response.getStatusLine().getStatusCode());
-            }
-  
-          } else {
-  
-            String auth = userName + ":" + accessKey;
-            String encodedAuth = "Basic " + Base64.getEncoder().encodeToString(auth.getBytes(StandardCharsets.UTF_8));
-  
-            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
-  
-              httpDelete.setHeader("Authorization", encodedAuth);
-  
-              int attempt = 0;
-              CloseableHttpResponse response = null;
-              while (attempt < 3) {
-                response = httpClient.execute(httpDelete);
-                if (response.getStatusLine().getStatusCode() == 200) {
-                  System.out.println("Tunnel stopped successfully");
-                  break;
-                } else {
-                  attempt++;
-                  if (response != null) response.close();
-  
-                  if (attempt == 3) {
-                    System.out.println("Max retries reached, could not stop tunnel");
-                    throw new TunnelException("Failed : HTTP error code : " + response.getStatusLine().getStatusCode());
-                  } else {
-                    Thread.sleep(3000);
-                  }
+            boolean stopTunnelAPI = Boolean.parseBoolean(System.getenv("STOP_TUNNEL_API"));
+
+            String deleteEndpoint = stopTunnelAPI ?
+                    "https://ts.lambdatest.com/v1.0/stop/" + tunnelID :
+                    "http://127.0.0.1:" + String.valueOf(infoAPIPortValue) + "/api/v1.0/stop";
+
+            HttpDelete httpDelete = new HttpDelete(deleteEndpoint);
+
+            if (!stopTunnelAPI) {
+
+                CloseableHttpClient httpclient = HttpClients.createDefault();

```

Comment:
the httpclient is not closed, can we use `try-with-resources` statement

Reply:
done